src/client/proxy/http.cc

```diff
@@ -422,9 +426,19 @@ void HTTPMessage::HandleJumpService() {
   // We should ask the user for confirmation before proceeding.
   // Previous reference: http://pastethis.i2p/raw/pn5fL4YNJL7OSWj3Sc6N/
   // We *could* redirect the user again to avoid dirtiness in the browser
-  kovri::client::context.GetAddressBook().InsertAddressIntoStorage(
-      m_Address, base64);
-  m_Path.erase(pos);
+  try
+  {
+    kovri::client::context.GetAddressBook().InsertAddressIntoStorage(
+        m_Address, base64);
```

Comment:
Also take a look inside this function. If the base64 can't be decoded, the function is simply returned. We want to either not catch the throw inside the function when base64 decoding fails - and assume caller responsibility (bad idea) or catch the whole function internally and re-throw (better idea).

Reply:
> Also take a look inside this function. If the base64 can't be decoded, the function is simply returned.

I separated base64 parsing + decoding into its own function, and will work on a PR to address the storage issue.